File: mobile/src/main/java/net/mm2d/dmsexplorer/util/update/UpdateInfo.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.util.update;

import com.squareup.moshi.Json;

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
public class UpdateInfo {
    private static class Mobile {
        @Json(name = "versionName")
        private String versionName;
        @Json(name = "versionCode")
        private int versionCode;
        @Json(name = "targetInclude")
        private int[] targetInclude;
        @Json(name = "targetExclude")
        private int[] targetExclude;
    }

    @Json(name = "mobile")
    private Mobile mobile;

    public boolean isValid() {
        return mobile != null
                && mobile.versionName != null
                && mobile.versionCode != 0
                && mobile.targetInclude != null
                && mobile.targetExclude != null;
    }

    public int getVersionCode() {
        return mobile.versionCode;
    }

    public String getVersionName() {
        return mobile.versionName;
    }

    public int[] getTargetInclude() {
        return mobile.targetInclude;
    }

    public int[] getTargetExclude() {
        return mobile.targetExclude;
    }
}

File: mobile/src/main/java/net/mm2d/dmsexplorer/util/update/UpdateInfo.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.util.update

import com.squareup.moshi.Json

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
internal data class UpdateInfo(
        @Json(name = "mobile")
        private val mobile: Mobile
) {
    val versionCode: Int
        get() = mobile.versionCode

    val versionName: String
        get() = mobile.versionName

    val targetInclude: List<Int>
        get() = mobile.targetInclude

    val targetExclude: List<Int>
        get() = mobile.targetExclude

    internal data class Mobile(
            @Json(name = "versionName")
            internal val versionName: String,
            @Json(name = "versionCode")
            internal val versionCode: Int,
            @Json(name = "targetInclude")
            internal val targetInclude: List<Int>,
            @Json(name = "targetExclude")
            internal val targetExclude: List<Int>
    )
}
